src/plugins/dashboard/public/dashboard_constants.ts

```diff
@@ -17,8 +19,12 @@ export const DashboardConstants = {
   SEARCH_SESSION_ID: 'searchSessionId',
 };
 
-export function createDashboardEditUrl(id: string) {
-  return `${DashboardConstants.VIEW_DASHBOARD_URL}/${id}`;
+export function createDashboardEditUrl(id?: string, editMode?: boolean) {
+  const edit = editMode ? `?${DASHBOARD_STATE_STORAGE_KEY}=(viewMode:edit)` : '';
```

Comment:
nit: you can use `setStateToKbnUrl` function from `kibana_utils` to set state in proper format

Reply:
I thought about using this, but it seems to prepend the entire current URL / not be used for routing within apps. I might be using it wrong though.

Reply:
I thought it should also work like this: 

```
setStateToKbnUrl('_a', {viewMode: 'edit'}, {storeInHashQuery: false}, '/dashboard/dashboard-id')
```

but I am not 100% sure now.  feel free to skip .

p.s. hate this weird fn signature 😅

Reply:
Yeah that did work, the trick was the `storeInHashQuery: false` argument. I will leave it in though it does seem to be adding quite a bit of complexity to this function.

Reply:
I actually had to revert this. Turns out importing the `stateToKbnUrl` method in that file causes a build failure in x-pack because of a functional test file using this. I ran into this before as well but forgot!